src/source.h

```diff
@@ -10,193 +10,119 @@ namespace cycamore {
+           " size.  It offers its material as a single commodity. If a composition\n" \
+           " recipe is specified, it provides that single material composition to\n" \
+           " requesters.  If unspecified, the source provides materials with the exact\n" \
+           " requested compositions.  The inventory size and throughput both default to\n" \
+           " infinite.  Supplies material results in corresponding decrease in\n" \
+           " inventory, and when the inventory size reaches zero, the source can provide\n" \
+           " no more material.\n" \
+  }
 
-  #pragma cyclus note {"doc": "A source facility that provides a " \
-                              "commodity with a given capacity"}
+  #pragma cyclus def clone
+  #pragma cyclus def schema
+  #pragma cyclus def annotations
+  #pragma cyclus def infiletodb
+  #pragma cyclus def snapshot
+  #pragma cyclus def snapshotinv
+  #pragma cyclus def initinv
 
-  ///   Print information about this agent
-  virtual std::string str();
-  // ---
+  virtual void InitFrom(Source* m);
 
-  // --- Agent Members ---
-  virtual void EnterNotify();
+  virtual void InitFrom(cyclus::QueryableBackend* b);
 
-  ///   Each facility is prompted to do its beginning-of-time-step
-  ///   stuff at the tick of the timer.
+  virtual void EnterNotify();
 
-  ///   @param time is the time to perform the tick
-  virtual void Tick();
+  virtual void Tick() {};
 
-  ///   Each facility is prompted to its end-of-time-step
-  ///   stuff on the tock of the timer.
+  virtual void Tock() {};
 
-  ///   @param time is the time to perform the tock
-  virtual void Tock();
+  virtual std::string str();
 
-  /// @brief Responds to each request for this source facility's commodity.
-  /// If a given request is more than this facility's capacity, it will offer
-  /// its capacity.
   virtual std::set<cyclus::BidPortfolio<cyclus::Material>::Ptr>
       GetMatlBids(cyclus::CommodMap<cyclus::Material>::type&
                   commod_requests);
 
-  /// @brief respond to each trade with a material made from this facility's
-  /// recipe
-  ///
-  /// @param trades all trades in which this trader is the supplier
-  /// @param responses a container to populate with responses to each trade
   virtual void GetMatlTrades(
     const std::vector< cyclus::Trade<cyclus::Material> >& trades,
     std::vector<std::pair<cyclus::Trade<cyclus::Material>,
     cyclus::Material::Ptr> >& responses);
-  // ---
-
-  // --- Source Members ---
-  ///   @brief creates a material object to offer to a requester
-  ///   @param target the material target a request desires
-  cyclus::Material::Ptr GetOffer(const cyclus::Material::Ptr target) const;
 
   ///   sets the output commodity name
   ///   @param name the commodity name
-  inline void commodity(std::string name) { out_commod = name; }
+  inline void commodity(std::string name) { outcommod = name; }
```

Comment:
Problem is that the member variables can't be named the same as the
functions.  So we either need to make the tests not use getters and then
remove them, rename the variables, or break the style guide - three
undesirables choices :/

On Wed, Apr 8, 2015 at 4:34 PM, Matthew Gidden <notifications@github.com>
wrote:

> In src/source.h
> <https://github.com/cyclus/cycamore/pull/310#discussion_r28020843>:
>
> >
> >    ///   sets the output commodity name
> >    ///   @param name the commodity name
> > -  inline void commodity(std::string name) { out_commod = name; }
> > +  inline void commodity(std::string name) { outcommod = name; }
>
> ah, it looks like they may be necessary for tests. they should match with
> gcsg then.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cycamore/pull/310/files#r28020843>.
>